impyute/dataset/base.py

```diff
@@ -65,6 +70,48 @@ def randn(theta=(0, 1), shape=(5, 5), missingness="mcar", thr=0.2, dtype="float"
     raw_data = getattr(corruptor, missingness)()
     return raw_data
 
+def randc(nlevels=5, shape=(5, 5), missingness="mcar", thr=0.2):
+    """ Return randomly generated dataset with uniformly distributed categorical data (alphabetic character)
+
+    Parameters
+    ----------
+    :param nlevels: int
+        Specify the number of different categories in the dataset
+    :param shape: tuple(optional)
+        Size of the randomly generated data
+    :param missingness: string in ('mcar', 'mar', 'mnar')
+        Type of missingness you want in your dataset
+    :param thr: float between [0,1]
+        Percentage of missing data in generated data
+    :return:
+    """
+    if shape[0]*shape[1] < nlevels:
+        raise BadInputError("nlevel exceeds the size of desired dataset. Please decrease the nlevel or increase the shape")
+
+    length = len(string.ascii_lowercase)
+    n_fold = math.floor(math.log(nlevels, length))
+    cat_pool = list(string.ascii_lowercase)
+
+    # when nlevel > 26, the alphabetical character is used up, need to generate extra strings as categorical data
+    if n_fold > 0:
+        for i in range(2, n_fold+2):
```

Comment:
This line seems to be causing a TypeError in the CI. `math.floor` returns an `int` only in Python 3. Should probably just add an explicit cast.

Reply:
Thanks for checking it!